pkg/component/shared/kubeapiserver.go

```diff
@@ -382,18 +383,49 @@ func ensureAdmissionPluginConfig(plugins []gardencorev1beta1.AdmissionPlugin) ([
 	return plugins, nil
 }
 
-func convertToAdmissionPluginConfigs(plugins []gardencorev1beta1.AdmissionPlugin) ([]kubeapiserver.AdmissionPluginConfig, error) {
-	var out []kubeapiserver.AdmissionPluginConfig
+func convertToAdmissionPluginConfigs(ctx context.Context, gardenClient client.Client, namespace string, plugins []gardencorev1beta1.AdmissionPlugin) ([]kubeapiserver.AdmissionPluginConfig, error) {
+	var (
+		kubeconfig []byte
+		err        error
+		out        []kubeapiserver.AdmissionPluginConfig
+	)
 
 	for _, plugin := range plugins {
-		out = append(out, kubeapiserver.AdmissionPluginConfig{
-			AdmissionPlugin: plugin,
-		})
+		if plugin.KubeconfigSecretName != nil {
+			key := client.ObjectKey{Namespace: namespace, Name: *plugin.KubeconfigSecretName}
+			kubeconfig, err = fetchKubeconfigFromSecret(ctx, gardenClient, key)
+			if err != nil {
+				return nil, fmt.Errorf("failed reading kubeconfig for webhook from referenced secret %s: %w", key, err)
+			}
+
+			out = append(out, kubeapiserver.AdmissionPluginConfig{
+				AdmissionPlugin: plugin,
+				Kubeconfig:      kubeconfig,
+			})
+		} else {
+			out = append(out, kubeapiserver.AdmissionPluginConfig{
+				AdmissionPlugin: plugin,
+			})
+		}
 	}
 
 	return out, nil
 }
 
+func fetchKubeconfigFromSecret(ctx context.Context, c client.Client, key client.ObjectKey) ([]byte, error) {
+	secret := &corev1.Secret{}
+	if err := c.Get(ctx, key, secret); err != nil {
+		return nil, err
+	}
+
+	kubeconfig, ok := secret.Data["kubeconfig"]
+	if !ok || len(kubeconfig) == 0 {
+		return nil, errors.New("the secret's field 'kubeconfig' is either not present or empty")
+	}
+
+	return kubeconfig, nil
+}
```

Comment:
This is the same function as in https://github.com/gardener/gardener/blob/32a8415cddb870771089f1f3ac08f0f4371b9179/pkg/operator/controller/garden/components.go#L545-L557. Can we make it reusable in a utils package instead of duplicating it?